### ISSUE TITLE
[CoE Starter Kit - BUG] Flow deleted from Power Automate but showing in the CoE report

### ISSUE THREAD

**Issue by PPTechy** [microsoft/coe-starter-kit#8866]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

Hi,

I have deleted flows using the make.powerautomate.com and make.powerapps.com, but those flows still showing in the CoE active flows report. How can we make this sync with CoE tool? If we delete a flow from an environment, that should not be shown in the CoE flow report. How can we fix this issue? If I am not wrong this flow "**Admin | Sync Template v4 (Flows)**" is responsible for this sync, if yes, this has been running successfully without error.

One observation, flow executed successfully in a few seconds:

<img width="714" alt="image" src="https://github.com/user-attachments/assets/84505c50-daef-4d74-baa8-168be861b532">

One more observation, if we fix the orphaned users permission using the "Manage Permissions" tool, again on the next run, I get the email with orphaned users which I already fixed and should not come in the email.

Hi @Jenefer-Monroe - I request your help here.

Thank you!

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.33

### What app or flow are you having the issue with?

Admin | Sync Template v4 (Flows)

### What method are you using to get inventory and telemetry?

None

**Comment by Jenefer-Monroe** [comment]
Hello sorry we will need to deal with these issues separately as having multiple issues in one posting makes tracking very difficult. So let's use this for your deleted issue. 

We do not check for object deletion every day due to the API impact of that. We have long running flows like this named with the prefix _CLEANUP -_ 
In particular, deletion is checked weekly by **_CLEANUP - Admin | Sync Template v4 (Check Deleted)_**
Please be sure it is turned on and running successfully. 
Below is more information about the system.

## Check Deleted explanation
Hello, please see below for an explanation of how we managed deleted objects in the CoE inventory

### Deleting objects from inventory
**How it works**
When objects are deleted from the tenant, they get marked deleted in the inventory when the following flow is run, along with its helpers: **_CLEANUP - Admin | Sync Template v4 (Check Deleted)_**
This is a long running flow and so it only runs once per week. 

Objects will stay in this state, marked as deleted from inventory, for ~2 months. This is to allow the record to be restored with its metadata in the event the object itself is restored in the tenant. 

At that point, once restore is no longer possible in the tenant, then the records themselves are deleted from the inventory tables as well. 

While they are in the "marked deleted" state they can be viewed in the kit using selecting that view for all the objects as shown here for apps.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/42ae9af2-6bb5-49fe-afba-838d5f925c68)

**Why keep them around for two months?** 
Here is the scenario we are supporting by keeping these around

- Flow X created in tenant
- Flow X added to the CoE inventory tables
- Flow X marked as exempt from inactivity processing in the CoE
- Flow X deleted from tenant
- Flow X marked as deleted in CoE inventory tables
- Flow X restored in tenant
- Flow X unmarked deleted in CoE inventory tables, exemption from inactivity processing remains in place.

**I need these records to stay in perpetuity**
We have an env var, _Also Delete from CoE_, which allows you to prevent the deletion from the inventory tables. Default behavior (yes) described above, but if you set it to no then these records will never be deleted by the kit

![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/af05bf15-e937-4032-9eab-e4d290108147)


**I don't want these records sitting around at all**
If you want the deletion from inventory to occur immediately, or at some other cadence like 6 months, then you will need to create your own flow on the side to do this. You can use the mechanism used in _CLEANUP - Admin | Sync Template v4 (Check Deleted)_ as a template for how to do this.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/187ec89f-77d7-48d4-8fb3-ca6df9b704d5)

**Comment by PPTechy** [comment]
Hi @Jenefer-Monroe,

Thanks a million much for your detailed explanation on this. I completely understand the concept of deletion.

As we know that this flow "**CLEANUP - Admin | Sync Template v4 (Check Deleted)**" runs once in a week, if I run this manually now, it will not delete the flows or apps, I just deleted few days back, it will only delete the 2 months older apps or flows. If my requirement is, I want to keep the CoE reports up to date and I don't want to wait for 2 months for the deletion, then, I need to create a copy of this flow "**CLEANUP - Admin | Sync Template v4 (Check Deleted)**" or create a new same flow referring to this flow, and there, I need update this flow configuration like below, from month to day in the "Time-Unit", then, all 2 days older deleted flows and apps will be deleted automatically from the inventory. Is my understanding correct?

<img width="463" alt="image" src="https://github.com/user-attachments/assets/f6dd9b3f-c2be-4e84-bf4c-a7fca4f1c622">

**Comment by Jenefer-Monroe** [comment]
Happy to help. 

Do note that all the reports and kit flows will filter our the items marked "deleted", so if all you are worried about is them showing up in reports, then you dont need to change this at all. 
Items that are marked deleted are hidden and there just in case they ever need restored.

If you want to delete them directly though, you are correct that is where you would update.

**Comment by PPTechy** [comment]
Hi @Jenefer-Monroe,

Thanks for your reply. I have enabled deleted column to the Deleted Flows report, I can see many flows with deleted checked mark. These flows we have deleted in the month of July, but the flows what I have deleted yesterday or two days back we don't see in the deleted flows report. Is it like these will be marked as deleted after the next run of this flow "**CLEANUP - Admin | Sync Template v4 (Check Deleted)**"?

Basically, what we need is - if the flows or apps deleted from the environment, immediately or at least in the next run this should be synced with the CoE report. How can we do that?

My next question is, if we deleted flow or app directly from the CoE report using the tool. Will it be deleted both from CoE report and Power Platform environment immediately? 

Thank you!

**Comment by Jenefer-Monroe** [comment]
### 1. Basically, what we need is - if the flows or apps deleted from the environment, immediately or at least in the next run this should be synced with the CoE report. How can we do that?

Unfortunately we do not have this. The API hit of checking for deleted objects is just too high for us to check daily so you can expect that to be weekly. You could change the check deleted flow to run daily instead of weekly, and see if it blows out your API too bad. If so you could add a process flow license to it. 

### 2. My next question is, if we deleted flow or app directly from the CoE report using the tool. Will it be deleted both from CoE report and Power Platform environment immediately?

Yes if you delete through our tools I believe we do mark the object as deleted while we are in there.

**Comment by PPTechy** [comment]
> ### 1. Basically, what we need is - if the flows or apps deleted from the environment, immediately or at least in the next run this should be synced with the CoE report. How can we do that?
> Unfortunately we do not have this. The API hit of checking for deleted objects is just too high for us to check daily so you can expect that to be weekly. You could change the check deleted flow to run daily instead of weekly, and see if it blows out your API too bad. If so you could add a process flow license to it.

The weekly runs will also be fine, if I delete a flow today from my environment, then when the next weekly "CLEANUP - "Admin | Sync Template v4 (Check Deleted)" flow runs, my today's deleted flow will be marked as deleted, and that will not be shown anymore in the active flows report, hence this marked deleted flows will not be included in the weekly admin email report. And the actual deletion will happen from the inventory after 2 months from the date marked as deleted? Is my understanding correct?

> 
> ### 2. My next question is, if we deleted flow or app directly from the CoE report using the tool. Will it be deleted both from CoE report and Power Platform environment immediately?
> Yes if you delete through our tools I believe we do mark the object as deleted while we are in there.

Thanks, I like this approach, I will rather prefer to delete the flows and apps directly from the CoE tool.

Hi @Jenefer-Monroe - I have added my comment, please have a look at these.

**Comment by Jenefer-Monroe** [comment]
Yes it sounds like your plan makes sense to me! And your understanding is correct.
Thanks for following up to let me know!

**Comment by PPTechy** [comment]
Hi @Jenefer-Monroe, Thank you so much. Please keep this thread open for the next couple of days, then we can close this.

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team
